api/swagger.yaml

```diff
@@ -77,43 +90,337 @@ tags:
-      | associateWithActorEvent         | ●        | ●       | ●        | ●     | ●     | ●     | `_PAR`          | Associate an entity with an `Actor`.
+      are the core of this API. Events link all static objects together. The sections below document the available event types. Each section contains
+      a table that summarizes to which entities the event can be applied and a table that denotes in which life cycle phases the event is available.
+
+      ## Update events
+
+      Update events typically send an update to the state of an entity. These kind of events are only sensible in the `actual` and `realized` life cycle
+      phases. This makes sense, because things like location or sensor updates can't be planned nor projected. Update events typically involve only one
+      entity at a time.
+
+      #### Applicable life cycle phases
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |         |           | ●      | ●        |
+
+      ### `locationUpdateEvent`
+
+      Updates the location of a `Vehicle`
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ### `sensorUpdateEvent`
+
+      Event to send an updated value of a sensor of the `Vehicle`. Can be used e.g. to monitor temperature inside a trailer.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          |         |          |       |       |       |            | ●      |       |
+
+      ### `startEngineEvent`, `stopEngineEvent`
+
+      The engine of a `Vehicle` is started or stopped.
+
+      #### Applicable entities
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ## Warning events
+
+      Warning events warn about violations, possible causes for delay or other things that may have impact on an operation and may need some
+      attention. Warning events can exist in all life cycle phases, since other events might be the cause of a warning.
+
+      #### Applicable life cycle phases
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |  ●      | ●         | ●      | ●        |
+
+      ### `restrictionWarningEvent`
+
+      A warning event that may be sent when a restriction is violated.
```

Comment:
Could this restriction warning event also be applicable to a vehicle? For example, an environment zone is passed where the vehicle is not allowed in a specific time window?

Reply:
Could we give an example for each entity type of what kind of restrictions we are talking about?